File: cpp/ball.cpp
#include "ball.h"

Ball::Ball()
{
}

Ball::~Ball()
{
}

void Ball::setRadius(float r)
{
    radius = r;
}

float Ball::getRadius()
{
    return radius;
}

float Ball::getWeight()
{
    return weight;
}

float Ball::getDiameter()
{
    return (2 * radius);
}

float Ball::getCircumference()
{
    return (2 * M_PI * radius);
}

float Ball::getArea()
{
    return (M_PI * pow(radius, 2));
}

float Ball::getVolume()
{
    return (4.0/3.0 * M_PI * pow(radius, 3.0));
}

void Ball::setWeight(float w)
{
    weight = w;
}

File: java/Ball.java
package my.languages;

public class Ball extends Sphere {
  private double radius;
  private double weight;
  
  public Ball() {
    System.out.println("[Create BALL]");
  }
  
  public Ball(double r, double w) {
    radius = r;
    weight = w;
  }
  
  public void setRadius(double r) {
    System.out.println("--> Setting radius for BALL");
    radius = r;
  }
  
  public void setWeight(double w) {
      weight = w;
  }
  
  public double getRadius() {
    return radius;
  }
  
  public double getDiameter() {
      return (2.0 * radius);
  }

  public double getCircumference() {
      return (2.0 * Math.PI * radius);
  }

  public double getArea() {
      return (4.0 * Math.PI * Math.pow(radius, 2.0));
  }
  
  public double getVolume() {
    return (4.0/3.0 * Math.PI * Math.pow(radius, 3.0));
  }
  
  public double getWeight() {
    return weight;
  }
}
